msprime/simulations.py

```diff
@@ -1204,6 +1217,95 @@ def __init__(self, population_size=1, psi=0.5, c=10.0):
         self.c = c
 
 
+class SingleSweep(SimulationModel):
+    # TODO document
+    name = "single_sweep"
+
+    def __init__(self, position, trajectory, population_size=1):
+        self.population_size = population_size
+        self.position = position
+        self.trajectory = trajectory
+
+    def get_ll_representation(self):
+        d = super().get_ll_representation()
+        d["position"] = self.position
+        d["time"] = self.trajectory.time
+        d["allele_frequency"] = self.trajectory.allele_frequency
+        return d
+
+
+class SweepTrajectory(object):
+    """
+    Class representing the allele frequency trajectory for a selective sweep.
+    The class has two attributes, time and allele_frequency which may be set
+    directly to define a trajectory.
+    """
+    def __init__(self, time=None, allele_frequency=None):
+        self.time = time
+        self.allele_frequency = allele_frequency
+
+
+class TrajectorySimulator(object):
+    """
+    Class to simulate an allele frequency trajectory on which to condition
+    the coalescent simulation.
+    """
+    def __init__(
+            self, start_frequency, end_frequency, alpha, time_slice, random_seed=None):
+        self.start_frequency = start_frequency
+        self.end_frequency = end_frequency
+        self.alpha = alpha
+        self.time_slice = time_slice
+        if random_seed is None:
+            random_seed = _get_random_seed()
+        self.rng = random.Random(random_seed)
+        self.__reset()
+
+    def __reset(self):
+        # TODO these should be numpy arrays
+        self.time = []
+        self.allele_frequency = []
+
+    def __genic_selection_stochastic_forwards(self, dt, freq, alpha):
+        ux = (alpha * freq * (1 - freq)) / np.tanh(alpha * freq)
+        sign = 1 if self.rng.random() < 0.5 else -1
+        freq += (ux * dt) + sign * np.sqrt(freq * (1.0 - freq) * dt)
+        return freq
+
+    def __simulate(self):
```

Comment:
we are going to want to have the population size history available within this routine. i guess it could come in through a DemographyDebugger object?